pkg/timelock/scheduler_test.go

```diff
@@ -153,6 +157,48 @@ func Test_dumpOperationStore(t *testing.T) {
 	assert.Equal(t, wantRead, gotRead)
 }
 
+func Test_scheduler_concurrency(t *testing.T) {
```

Comment:
After the deadlocks I saw with the earlier version, I decided to add this unorthodox test to validate that the scheduler can simultaneously add and consume operations.